packages/core/src/state-management/repository-internals.ts

```diff
@@ -244,6 +239,14 @@ export class RepositoryInternals {
     return this.fromStreamStateStore(streamId)
   }
 
+  async fromMemoryOrStoreSafe(streamId: StreamID): Promise<RunningState | undefined> {
```

Comment:
please, comment what makes this function "safe" compared to regular `fromMemoryOrStore`.

Also, can this be implemented in terms of `fromMemoryOrStore` instead of calling _fromMemory and fromStreamStateStore directly?

Reply:
Actually, yes. Done.